auth_service/auth_adapter/core/auth.py

```diff
@@ -18,121 +18,207 @@
-        try:
-            if not external_keys:
-                raise ValueError("No external keys configured")
-            self.external_jwks = jwk.JWKSet.from_json(external_keys)
-        except Exception as exc:  # pylint:disable=broad-except
-            # do not throw an error so that the auth adapter can still run
-            # even though it will not be able to validate external tokens
-            log.error("Error in external signing keys: %s", exc)
+        if external_keys:
+            try:
+                self.external_jwks = jwk.JWKSet.from_json(external_keys)
+            except Exception as exc:  # pylint:disable=broad-except
+                # do not throw an error so that the auth adapter can still run
+                # even though it will not be able to validate external tokens
+                log.error("Cannot parse external signing keys: %s", exc)
+        else:
+            log.error("No external signing keys configured.")
         internal_keys = config.auth_int_keys
-        try:
-            if not internal_keys:
-                raise ValueError("No internal signing keys configured.")
-            self.internal_jwk = jwk.JWK.from_json(internal_keys)
-        except Exception as exc:  # pylint:disable=broad-except
-            # do not throw an error so that the auth adapter can still run
-            # even though it will not be able to sign internal tokens
-            log.error("Error in internal signing key pair: %s", exc)
-
-
-signing_keys = SigningKeys()
+        if internal_keys:
+            try:
+                self.internal_jwk = jwk.JWK.from_json(internal_keys)
+            except Exception as exc:  # pylint:disable=broad-except
+                # do not throw an error so that the auth adapter can still run
+                # even though it will not be able to sign internal tokens
+                log.error("Cannot parse internal signing key pair: %s", exc)
+        else:
+            log.error("No internal signing keys configured.")
+        external_algs = config.auth_ext_algs
+        if external_algs:
+            self.external_algs = external_algs
+        else:
+            log.warning("Allowed external signing algorithms not configured.")
+        authority_url = config.oidc_authority_url
+        if authority_url:
+            self.check_claims["iss"] = authority_url
+        else:
+            log.warning("No OIDC authority URL configured.")
+        client_id = config.oidc_client_id
+        if client_id:
+            self.check_claims["client_id"] = client_id
+        else:
+            log.warning("No OIDC client ID configured.")
+
+
+jwt_config = JWTConfig()
+
+
+def _compare_user(user: User, external_claims: dict[str, Any]) -> Optional[str]:
+    """Compare user with external claims and return inactivation context."""
+    if user.status == UserStatus.ACTIVATED:
+        if user.name != external_claims.get("name"):
+            return "name change"
```

Comment:
Might be a bit nicer to simply raise a custom exception here.

Reply:
Changed in 8fe39e9.